test/v1/controllers/DeleteOtherExpensesControllerSpec.scala

```diff
@@ -0,0 +1,143 @@
+  private val correlationId = "X-123"
+
+  private val rawData = DeleteOtherExpensesRawData(nino, taxYear)
+  private val requestData = DeleteOtherExpensesRequest(Nino(nino), taxYear)
+
+  "handleRequest" should {
+    "return NoContent" when {
+      "the request recieved is valid" in new Test {
+
+        MockDeleteOtherExpensesRequestDataParser
+          .parse(rawData)
+          .returns(Right(requestData))
+
+        MockDeleteOtherExpensesService
+          .delete(requestData)
+          .returns(Future.successful(Right(ResponseWrapper(correlationId, ()))))
+
+        val result: Future[Result] = controller.handleRequest(nino, taxYear)(fakeRequest)
+
+        status(result) shouldBe NO_CONTENT
+        header("X-CorrelationId", result) shouldBe Some(correlationId)
+      }
+    }
+    "return the error as per spec" when {
+      "parser errors occur" should {
+        def errorsFromParserTester(error: MtdError, expectedStatus: Int): Unit = {
+          s"a ${error.code} error is returned from the parser" in new Test {
+
+            MockDeleteOtherExpensesRequestDataParser
+              .parse(rawData)
+              .returns(Left(ErrorWrapper(Some(correlationId), error, None)))
+
+            val result: Future[Result] = controller.handleRequest(nino, taxYear)(fakeRequest)
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(error)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (BadRequestError, BAD_REQUEST),
+          (NinoFormatError, BAD_REQUEST),
+          (TaxYearFormatError, BAD_REQUEST),
+          (RuleTaxYearRangeInvalidError, BAD_REQUEST)
+        )
+
+        input.foreach(args => (errorsFromParserTester _).tupled(args))
+      }
+
+      "service errors occur" should {
+        def serviceErrors(mtdError: MtdError, expectedStatus: Int): Unit = {
+          s"a $mtdError error is returned from the service" in new Test {
+
+            MockDeleteOtherExpensesRequestDataParser
+              .parse(rawData)
+              .returns(Right(requestData))
+
+            MockDeleteOtherExpensesService
+              .delete(requestData)
+              .returns(Future.successful(Left(ErrorWrapper(Some(correlationId), mtdError))))
+
+            val result: Future[Result] = controller.handleRequest(nino, taxYear)(fakeRequest)
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(mtdError)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+          }
+        }
+
+        val input = Seq(
+          (NinoFormatError, BAD_REQUEST),
+          (TaxYearFormatError, BAD_REQUEST),
+          (RuleTaxYearRangeInvalidError, BAD_REQUEST),
+          (NotFoundError, NOT_FOUND),
+          (DownstreamError, INTERNAL_SERVER_ERROR),
+          (UnauthorisedError, FORBIDDEN)
+        )
```

Comment:
Service error only should include des errors as these are backend errors

RuleTaxYearRangeInvalidError shouldn't be there as well as UnauthorisedError